src/Appwrite/Extend/Exception.php

```diff
@@ -298,11 +301,20 @@ class Exception extends \Exception
     protected array $errors = [];
     protected bool $publish;
 
-    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int $code = null, \Throwable $previous = null)
+    public function __construct(string $type = Exception::GENERAL_UNKNOWN, string $message = null, int|string $code = null, \Throwable $previous = null)
     {
         $this->errors = Config::getParam('errors');
         $this->type = $type;
         $this->code = $code ?? $this->errors[$type]['code'];
+
+        if(\is_string($this->code)) {
```

Comment:
@Meldiron lets add a comment here to indicate this is done for SQL errors that are usually strings like `HY001` etc.

Reply:
Will do, in which branch?

Reply:
in the current branch